mlos_core/mlos_core/optimizers/utils.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Contains utils used for implementing the mlos_core optimizers
+"""
+import inspect
+from typing import Any, Callable, Dict, List, Optional
+import pandas as pd
+
+
+def to_metadata(metadata: Optional[pd.DataFrame]) -> Optional[List[pd.Series]]:
+    """
+    Converts a list of metadata dataframe objects to a list of metadata objects
+    Parameters
+    ----------
+    metadata : Optional[pd.DataFrame]
+        The dataframe to convert to metadata
+
+    Returns
+    -------
+    Optional[List[pd.Series]]
+        The created metadata object
+    """
+    if metadata is None:
+        return None
+    return [idx_series[1] for idx_series in metadata.iterrows()]
+
+
+def filter_kwargs(function: Callable, **kwargs: Any) -> Dict[str, Any]:
```

Comment:
Don't we have this somewhere else as well?

Reply:
This used to be in the smac_optimizer.py file, but now lives here based on prior comments

Reply:
I think I was thinking of `_get_kwarg_defaults` but that's a little different and in a different module.